runtime/parachains/src/inclusion.rs

```diff
@@ -193,35 +220,90 @@ impl<T: Trait> Module<T> {
 			}
 		}
 
-		// TODO [now] actually apply bitfields
+		let now = <system::Module<T>>::block_number();
+		for signed_bitfield in signed_bitfields.0 {
+			for (bit_idx, _)
+				in signed_bitfield.bitfield.0.iter().enumerate().filter(|(_, is_av)| **is_av)
+			{
+				let record = assigned_paras_record[bit_idx]
+					.as_mut()
+					.expect("validator bitfields checked not to contain bits corresponding to unoccupied cores; qed");
+
+				// defensive check - this is constructed by loading the availability bitfield record,
+				// which is always `Some` if the core is occupied - that's why we're here.
+				let val_idx = signed_bitfield.validator_index as usize;
+				if let Some(mut bit) = record.1.as_mut()
+					.and_then(|r| r.availability_votes.get_mut(val_idx))
+				{
+					*bit = true;
+				}
```

Comment:
```suggestion
				} else if cfg!(debug_assertion) { unreachable!("Core must be occupied to come here. qed") }
```
assert on debug builds, so miss-usage is detected in debug cases but are being let slide in release

Reply:
Why let them slide on release?

Reply:
Nevermind, of course we don't want to panic in release code. I misread, thought it would cause a compile-time failure, not a runtime. This is a good idea.